android/app/src/main/java/org/fossasia/openevent/data/Event.kt

```diff
@@ -1,62 +1,93 @@
 package org.fossasia.openevent.data
 
+import com.fasterxml.jackson.annotation.JsonProperty
 import com.fasterxml.jackson.databind.PropertyNamingStrategy
 import com.fasterxml.jackson.databind.annotation.JsonNaming
 import com.github.jasminb.jsonapi.IntegerIdHandler
 import com.github.jasminb.jsonapi.annotations.Id
 import com.github.jasminb.jsonapi.annotations.Relationship
 import com.github.jasminb.jsonapi.annotations.Type
-
-import org.fossasia.openevent.data.extras.Copyright
-import org.fossasia.openevent.data.extras.SocialLink
-import org.fossasia.openevent.data.extras.SpeakersCall
-
 import io.realm.RealmList
 import io.realm.RealmObject
 import io.realm.annotations.PrimaryKey
+import org.fossasia.openevent.data.extras.Copyright
+import org.fossasia.openevent.data.extras.SocialLink
+import org.fossasia.openevent.data.extras.SpeakersCall
 
 @Type("event")
-@JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
```

Comment:
Nope, only use JsonProperty for ambiguous fields. This is boilerplate. We don't want to write extra code which needs to be maintained. This is like going back in time/worse code

Reply:
Yeah but tests fail becuase it says that if one property is annotated, all the properties have to be. I tried using @JsonProperty() at just the required fields and using the namingStrategy for others but it doesn't work that way

Reply:
Then I'm afraid that this solution doesn't work. This is the worst boilerplate and we have had a lot of problems related to this in the past. This is type unsafe and a property change requires 2 changes. 

I'll look for another solution but if there is none, I'm afraid that this trade off will be too huge to make. We'll have to look into ways of mixing in lombok and kotlin

Reply:
https://stackoverflow.com/a/44440741/3309666

Just use getter annotation

Reply:
Secondly, `@JsonProperty` works as expected

Use it instead
`@field:JsonProperty("is-admin")`